Assignment4/LauraRamirezM-005955793/task07.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1E_EJtv2DVYLx83JsIjGbqy9CCbMbRLD5
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+"""First let's read the RDF file"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+# TO DO
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+
+# TO DO in RDFLib
+print("RDFLib")
+for s,p,o in g.triples((None,RDFS.subClassOf,ns.LivingThing)):
+  print(s)
+
+# TO DO in SPARQL
+print("SPARQL")
+q1 = prepareQuery('''
+  SELECT ?Subject WHERE {
+    ?Subject RDFS:subClassOf ns:LivingThing.
+  }
+  ''',
+  initNs = {"RDFS": RDFS, "ns": ns}
+)
+
+# Visualize the results
+for r in g.query(q1):
+  print(r.Subject)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+# TO DO
+
+# TO DO in RDFLib
+print("RDFLib")
+for s,p,o in g.triples((None,RDFS.subClassOf,ns.Person)):
+  for s1,p1,o1 in g.triples((None,RDF.type,s)):
+    print(s1)
+
+for s2,p2,o2 in g.triples((None,RDF.type,ns.Person)):
+  print(s2)
```

Comment:
This will not print out all the subclasses of subclasses of person. Try doing the function recursively